oat-sa/tao-core
# Feature/AUT-3627/Apply the Solar Design to the header/footer bar

### Related to: https://oat-sa.atlassian.net/browse/AUT-3646

### Summary

Apply the Solar Design to both the header and the footer bars.

Out-of-scope:
- the taste queue
- menu hiding

![image](https://github.com/oat-sa/tao-core/assets/1500098/ee5a106f-c71b-4326-b2ff-cf5697fb4023)

### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

The base theme for the header/footer bars has been slightly refactored to support the use of such variables.

### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3627/solar-design-header-bar`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the header/fo0ter bars, compare with the UX mockups
